src/value.rs

```diff
@@ -215,6 +217,40 @@ impl Uninitialized for Value {
     }
 }
 
+impl FromGlibPtrNone<*mut gobject_ffi::GValue> for Value {
+    #[inline]
+    unsafe fn from_glib_none(ptr: *mut gobject_ffi::GValue) -> Self {
+        assert!(!ptr.is_null());
+        Value(ptr::read(ptr))
+    }
+}
+
+impl FromGlibPtrFull<*mut gobject_ffi::GValue> for Value {
+    #[inline]
+    unsafe fn from_glib_full(ptr: *mut gobject_ffi::GValue) -> Self {
+        let value = Value(ptr::read(ptr));
+        gobject_ffi::g_value_unset(ptr);
+        value
```

Comment:
This one doesn't have the above problem, because the C code passed you ownership of whatever the GValue contains, but it also passed you ownership of the GValue itself. You need to free it afterwards, or leak memory.